Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM.cpp

```diff
@@ -184,9 +234,9 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian(
                         // H_exchange
                         if (mag_exchange_arrx == 0._rt) amrex::Abort("The mag_exchange_arrx is 0.0 while including the exchange coupling term H_exchange for H_eff");
                         amrex::Real const H_exchange_coeff = 2.0 * mag_exchange_arrx / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx;
-                        Hx_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0);
-                        Hy_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1);
-                        Hz_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2);
+                        Hx_eff += H_exchange_coeff * LapM_old_xface(i, j, k, 0);
+                        Hy_eff += H_exchange_coeff * LapM_old_xface(i, j, k, 1);
+                        Hz_eff += H_exchange_coeff * LapM_old_xface(i, j, k, 2);
```

Comment:
I think the code will be more concise without MultiFab `LapM_old`. The change will be:
1. Delete Line 102-147 and Line 179-182;
2. Change Line 237-239 to 
```
Hx_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface_old, coefs_x, coefs_y, coefs_z, i, j, k, 0);
Hy_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface_old, coefs_x, coefs_y, coefs_z, i, j, k, 1);
Hz_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface_old, coefs_x, coefs_y, coefs_z, i, j, k, 2);
```

Reply:
In the first order case, yes this is better since (a) it saves memory, and (b) LapM_old is only needed once.  In the second-order scheme we either need to keep LapM_old around in memory OR recompute it each iteration, so the choice is less clear.